pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -479,31 +479,53 @@ func streamPathToBuild(git git.Repository, in io.Reader, out io.Writer, client o
 				if gitErr != nil {
 					return nil, fmt.Errorf("the directory %q is not a valid Git repository: %v", clean, gitErr)
 				}
-				pr, pw := io.Pipe()
-				go func() {
-					if err := git.Archive(clean, options.Commit, "tar.gz", pw); err != nil {
-						pw.CloseWithError(fmt.Errorf("unable to create Git archive of %q for build: %v", clean, err))
-					} else {
-						pw.CloseWithError(io.EOF)
-					}
-				}()
-				r = pr
+
+				// Create a temp directory to move the repo contents to
+				tempDirectory, err := ioutil.TempDir(os.TempDir(), "oc_cloning_"+options.Commit)
+				if err != nil {
+					return nil, err
+				}
+				defer os.RemoveAll(tempDirectory)
+
+				// We only want to grab the contents of the specified commit, with
+				// submodules included
+				cloneOptions := git.CloneOptions{
+					Recursive: true,
+					Shallow:   false,
+					// TODO: Should we output this at a loglevel?  If so, what level?
+					Quiet: true,
+				}
+
+				// Clone the repository to a temp directory for future tar-ing
+				if err := repo.CloneWithOptions(tempDirectory, clean, cloneOptions); err != nil {
+					return nil, err
+				}
+				if err := repo.Checkout(tempDirectory, options.Commit); err != nil {
+					return nil, err
+				}
+				if err := os.RemoveAll(filepath.Join(tempDirectory, ".git")); err != nil {
+					return nil, err
+				}
```

Comment:
>  Is the defer RemoveAll on line 488 not cleaning the temp directory?

yes it is, sorry my bad.  Saw the removall down here and assumed that was how you were cleaning up, didn't read it carefully enough.